### ISSUE TITLE
Ship of Harkinian - Ocarina of Time port [$685]

### ISSUE THREAD

**Issue by Meh99k** [vita-nuova/bounties#118]
I dont know much when it comes to coding vita ports, but I was hoping the Vita could get some form of an Ocarina of Time port.
- It's been fully decompiled on pc
- Ship of Harkinian is a good fan project that builds upon the original decompilation and has a very active community, it's even been ported to the switch.

I opened an account to make this issue so bear with me I'm very new to this lol. 

I'm not sure how to open a bounty on this post but if anyone could assist me with that, it would be greatly appreciated.

**Comment by Rinnegatamante** [comment]
https://app.bountysource.com/?order=created_at&direction=desc You need to create an account here and then click on Post Bounty and insert this issue as linked issue.

**Comment by Meh99k** [comment]
Alright I'll get to that soon, thanks

**Comment by kworrell13** [comment]
While waiting for this to be ported, you could try emulating it. The vita emulates Ocarina of Time quite well using the daedalusX64 emulator.

**Comment by Meh99k** [comment]
> While waiting for this to be ported, you could try emulating it. The vita emulates Ocarina of Time quite well using the daedalusX64 emulator.

Ah yeah I did actually try that out, unfortunately the gameplay was pretty choppy and overall it wasn't that good of experience on my end even after tinkering with the settings. Thanks for the suggestion though. Hopefully when it's executed natively it'll run better lol

**Comment by Akeruz** [comment]
Yes

**Comment by Akeruz** [comment]
I know a Lil of devmode how can i port it ?

**Comment by BlackSheepBoy69** [comment]
[+$120]

There didn't seem to be any interest in my bounty from ~1 year ago. I've emailed bountysource to have funds from that bounty transferred here. ~~If the Zelda port gets done before the transfer happens, claim this bounty with it: https://github.com/vita-nuova/bounties/issues/86~~

**Comment by BlackSheepBoy69** [comment]
Ok it's showing up now on the bountysource page. It was glitched for a second there. @Rinnegatamante  might need to update the bounty amount manually since it was glitched

**Comment by nngeorgiy** [comment]
I'll take this, if you don't mind. The plan is to do a honest Shipwright port to the standards of the existing WiiU one, with Vita-specific graphics and input systems, then hopefully upstream it. A lot depends on maintainers' willingness, so I'll tell you for sure if I'll be able to see this through early next month.

**Comment by Rinnegatamante** [comment]
I've already it running on Vita. Just waiting for a specific bug to be solved on upstream.
https://www.youtube.com/watch?v=_oo8M7wlZN8

**Comment by nngeorgiy** [comment]
Looks good! Thanks for the heads-up. I'll move on to greener pastures then.

**Comment by mattbell87** [comment]
That looks awesome @Rinnegatamante. True to the OG framerate I see. Thankyou for all of your Vita ports! 💪

**Comment by Chubbles64** [comment]
Any update on this @Rinnegatamante?

**Comment by Rinnegatamante** [comment]
Nope, as of today the bug present on upstream making Vita build crash shortly after booting the game is still there.

**Comment by alastair-warren** [comment]
@Rinnegatamante which bug is it out of interest? I'd be interested to follow it.

Thank you for all the work you've done on the vita scene too

**Comment by Rinnegatamante** [comment]
There's some kernel objects leak (mutex/sema/thread handles, etc) that makes the game crash shortly after launching a new game.

**Comment by AntonIXO** [comment]
> There's some kernel objects leak (mutex/sema/thread handles, etc) that makes the game crash shortly after launching a new game.

Could you share source code, please?

**Comment by Rinnegatamante** [comment]
Most of my WiP stuffs are on my profile https://github.com/Rinnegatamante/Shipwright/tree/upstream-develop

**Comment by drewdrew21b** [comment]
> There's some kernel objects leak (mutex/sema/thread handles, etc) that makes the game crash shortly after launching a new game.

Just curious, was this reported in the Issues tracker for the primary repository? https://github.com/HarbourMasters/Shipwright/issues?page=1&q=is%3Aissue+is%3Aopen

I want to follow it so I can see updates around this.

Also, you have done an amazing job with the countless other ports for Vita, so can't wait to see if this comes to fruition eventually!

**Comment by Rinnegatamante** [comment]
I didn't create an Issue on main repo but the team knows about the problematic.

**Comment by Timmy1st** [comment]
Any update on the upstream bug @Rinnegatamante

**Comment by Newah1** [comment]
Any progress on this?

**Comment by BlackSheepBoy69** [comment]
~~With the bounty being this high that I'm sure if there was updates, we'd know. The issue is in the Ship of Harkinian code, not Rinne's it seems like. Someone will need to bugfix the massive Ship of Harkinian code which is probably a lot harder than porting it and is better left off to people actually familiar with the Ship of Harkinian code. There is literally 300+ currently open issues with the current Ship of Harkinian code, so I'm not sure how fast or easy bugfixing any of that will be.~~

~~Someone could, for example, open a separate bounty to fix the actual Ship of Harkinian code issue preventing Rinne's port~~

**Comment by Rinnegatamante** [comment]
Bug in upstream may be fixed but currently it's impossible to test if it's indeed fixed since downstreamed version just crashes prior even mounting up the OTR file.

**Comment by fox87111** [comment]
So the Harbourmaster Team is responsible for the upstream bug mentioned before. Who is responsible for this downstream bug Rinne mentioned?

**Comment by Rinnegatamante** [comment]
Shipwright simply is too fat for Vita it seems. DaedalusX64 runs the game way bette than Shipwright does. I've also tried to optimize it any further (introducing NEON usage wherever possible, moving to faster rendering pipelines more optimized for sceGxm architecture and some other minor adjustments to improve framerate (eg: optimizing the texture cache lookup code) however the game still has sections in which framerate lowers quite a while.

Currently waiting for a project worked on by another developer that may bring some backportable improvements to Shipwright as well although I've not high hopes for this to be fullspeed.

To the backers of the bounty: I don't even know if it's worth releasing this in its current state. It's on average playable but honestly DaedalusX64 offers a better experience overall. If you guys want to get back your money, you can submit a request to Bountysource or you can all agree to shifting the bounty onto another target if you don't think its state is worth the money.

Last update: https://twitter.com/Rinnegatamante/status/1687492209577279493

**Comment by Chubbles64** [comment]
Probably not the right place to have this discussion so I apologize, but I would still like to see this released. I know it's still in a rough shape judging by the video, but I'm still interested and would be willing to leave my money on this bounty.
